client/client-lib/src/lib.rs

```diff
@@ -833,6 +833,11 @@ impl Client<UserClientConfig> {
 
         let consensus_height = self.context.api.fetch_consensus_block_height().await?;
         let absolute_timelock = consensus_height + OUTGOING_LN_CONTRACT_TIMELOCK;
+        let consensus_clock_time = self.context.api.fetch_consensus_clock_time().await?;
+
+        if invoice.would_expire(consensus_clock_time.duration_since(UNIX_EPOCH).unwrap()) {
+            return Err(ClientError::LnClientError(LnClientError::ExpiredInvoice));
+        }
```

Comment:
For outgoing LN payments the recipient will tell the gateway that the invoice expired, which triggers a refund. We don't need to pre-check that necessarily. On the incoming side where we are the recipient we should do that on the other hand.